bitcampHomework17/src/ba/bitcamp/homework17/part01/task04/Client.java

```diff
@@ -0,0 +1,88 @@
+package ba.bitcamp.homework17.part01.task04;
+
+import ba.bitcamp.homework17.part01.task01.Computer;
+import ba.bitcamp.homework17.part01.task02.Network;
+
+public class Client extends Computer implements Connectable {
+
+	private Computer connectedWith;
+	private Network whichNetworkConnected;
+
+	/**
+	 * <h1>Creating constructor that initializes computer name and mac address.</h1>
+	 * 
+	 * @param name
+	 *            - represents computer name (inherited from class Computer)
+	 * @param macAddress
+	 *            - represents mac address (inherited from class Computer)
+	 */
+	public Client(String name, char[] macAddress) {
+		super(name, macAddress);
+	}
+
+	// Getters
+	public Computer getConnectedWith() {
+
+		return connectedWith;
+	}
+
+	public Network getWhichNetworkConnected() {
+		return whichNetworkConnected;
+	}
+
+	// Method that connects two computers
+	@Override
+	public void connect(Computer c) {
+		if (connectedWith != null || whichNetworkConnected != null) {
+			throw new IllegalArgumentException("Error");
```

Comment:
Poruka moze biti opisnija, npr. "Error while connecting to computer. Link to computer or network already exists."